pysite/views/api/bot/tag.py

```diff
@@ -1,66 +1,56 @@
 # coding=utf-8
 
-from flask import g, jsonify, request
+from flask import jsonify, request
 
-import rethinkdb
-
-from pysite.base_route import APIView
+from pysite.base_route import APIView, DBViewMixin
 from pysite.constants import ErrorCodes
 
 
-class TagView(APIView):
-    path = '/tag'
-    name = 'tag'
-    table = 'tag'
-
-    def __init__(self):
-        # make sure the table exists
-        with g.db.get_connection() as conn:
-            try:
-                rethinkdb.db(g.db.database).table_create(self.table, {'primary_key': 'tag_name'}).run(conn)
-            except rethinkdb.RqlRuntimeError:
-                print(f'Table {self.table} exists')
+class TagView(APIView, DBViewMixin):
+    path = "/tag"
+    name = "tag"
+    table_name = "tag"
+    table_primary_key = "tag_name"
 
     def get(self):
         """
-        Indata must be provided as params,
+        Data must be provided as params,
         API key must be provided as header
         """
-        rdb = rethinkdb.table(self.table)
-        api_key = request.headers.get('X-API-Key')
-        tag_name = request.args.get('tag_name')
+        api_key = request.headers.get("X-API-Key")
+        tag_name = request.args.get("tag_name")
 
         if self.validate_key(api_key):
             if tag_name:
-                data = rdb.get(tag_name).run(g.db.conn)
-                data = dict(data) if data else {}
+                data = self.db.get(self.table_name, tag_name)
             else:
-                data = rdb.pluck('tag_name').run(g.db.conn)
-                data = list(data) if data else []
+                data = self.db.pluck(self.table_name, "tag_name")
         else:
             return self.error(ErrorCodes.invalid_api_key)
 
-        return jsonify(data)
+        return jsonify(data if data is not None else {})
```

Comment:
for this situation I kinda like 
```python
return jsonify(data or {})
```
it's less explicit, but it also reads really well.